pyuntl/untldoc.py

```diff
@@ -675,3 +676,42 @@ def find_untl_errors(untl_dict, **kwargs):
         'error_dict': error_dict,
     }
     return found_data
+
+
+def untl_dict_to_tuple(untl_dict):
+    """Convert values of untl dict to tuple form which will be later used for hashing."""
+    untl_tuple = {}
+    for elem in untl_dict:
+        for i, v in enumerate(untl_dict[elem]):
+            if isinstance(untl_dict[elem][i]['content'], dict):
+                untl_dict[elem][i]['content'] = [v for v in untl_dict[elem][i]['content'].items()]
+                untl_dict[elem][i]['content'].sort()
+            untl_dict[elem][i] = [v for v in untl_dict[elem][i].items()]
+        untl_tuple[elem] = untl_dict[elem]
+    return untl_tuple
+
+
+def generate_hash(input):
+    """Return an md5 hash of the output of untl_to_tuple_form."""
+    return hashlib.md5(repr(input).encode()).hexdigest()
+
+
+def untl_to_hash_dict(untl_elements, meaningfulMeta=True):
+    """Convert untl element to hashed untl dictionary."""
+    if meaningfulMeta:
+        remove_elements = []
+        for element in untl_elements.children:
+            if (
+                element.tag == 'meta'
+                and (element.qualifier == 'metadataModificationDate'
+                     or element.qualifier == 'metadataModifier')
+            ):
+                remove_elements.append(element)
+        for elem in remove_elements:
+            untl_elements.remove_child(elem)
+    untl_dict = untlpy2dict(untl_elements)
+    untl_tuple = untl_dict_to_tuple(untl_dict)
+    hash_dict = {}
+    for tup in untl_tuple:
+        hash_dict[tup] = generate_hash(untl_tuple[tup])
```

Comment:
Perhaps this could be of the pattern `for key, value in untl_tuple.items(): hash_dict[key] = generate_hash(value)`?